### PR TITLE
Respect debugger suggestion mode settings

### PR DESCRIPTION
**Pull request by rchande**
Start completion in the watch/immediate windows. If the user toggles suggestion mode off in the watch/immediate windows, leave it off.

### CODE REVIEW

**Comment by rchande** [comment]
Addresses the rest of https://github.com/dotnet/roslyn/issues/10343#issuecomment-273479661
Tag @dotnet/roslyn-ide for review

File: src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_ToggleCompletionMode.cs [hunk 1]
```diff
@@ -23,8 +24,18 @@ void ICommandHandler<ToggleCompletionModeCommandArgs>.ExecuteCommand(ToggleCompl
         {
             if (Workspace.TryGetWorkspace(args.SubjectBuffer.AsTextContainer(), out var workspace))
             {
-                var newState = !workspace.Options.GetOption(EditorCompletionOptions.UseSuggestionMode);
-                workspace.Options = workspace.Options.WithChangedOption(EditorCompletionOptions.UseSuggestionMode, newState);
+                Option<bool> option;
+                if (_isDebugger)
```
Comment (CyrusNajmabadi) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_ToggleCompletionMode.cs]: user ?: expression.

File: src/EditorFeatures/Core/Options/CompletionOptions.cs [hunk 2]
```diff
@@ -10,5 +10,10 @@ internal static class EditorCompletionOptions
 
         [ExportOption]
         public static readonly Option<bool> UseSuggestionMode = new Option<bool>(FeatureName, nameof(UseSuggestionMode), defaultValue: false);
+
+        // Default into suggestion mode in the watch/immediate windows but respect the
+        // user's preferences if they switch away from it.
+        [ExportOption]
+        public static readonly Option<bool> UseSuggestionMode_Debugger = new Option<bool>(FeatureName, nameof(UseSuggestionMode_Debugger), defaultValue: true);
```
Comment (jasonmalinowski) [inline src/EditorFeatures/Core/Options/CompletionOptions.cs]: The intent is this isn't persisted somewhere? i.e. next VS run this is back to what it was before?

(Not applying judgement, just confirming intent.)

File: src/VisualStudio/Core/Test/DebuggerIntelliSense/CSharpDebuggerIntellisenseTests.vb [hunk 3]
```diff
@@ -667,6 +667,36 @@ $$</Document>
...
+{
+    static void Main(string[] args)
+    [|{|]
+
+    }
+}</Document>
+                           </Project>
+                       </Workspace>
+
+            Using state = TestState.CreateCSharpTestState(text, False)
```
Comment (CyrusNajmabadi) [inline src/VisualStudio/Core/Test/DebuggerIntelliSense/CSharpDebuggerIntellisenseTests.vb]: what is 'false' here?

File: src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_ToggleCompletionMode.cs (hunk 1 above)
Comment (jasonmalinowski) [inline src/EditorFeatures/Core/Implementation/IntelliSense/Completion/Controller_ToggleCompletionMode.cs]: ternary operator?

File: src/EditorFeatures/Core/Options/CompletionOptions.cs (hunk 2 above)
Comment (rchande) [inline src/EditorFeatures/Core/Options/CompletionOptions.cs]: Right, just like non-debugger suggestion mode.

**Comment by CyrusNajmabadi** [comment]
Can you please add a comment making that clear.  In the future someone may think its unintentional that we do not persist these options.

File: src/VisualStudio/Core/Test/DebuggerIntelliSense/CSharpDebuggerIntellisenseTests.vb (hunk 3 above)
Comment (rchande) [inline src/VisualStudio/Core/Test/DebuggerIntelliSense/CSharpDebuggerIntellisenseTests.vb]: Controls whether the buffer is set up as a watch window or immediate window. Separated out a test for each of those.

File: src/Features/Core/Portable/Completion/SuggestionMode/SuggestionModeCompletionProvider.cs [hunk 4]
```diff
@@ -15,16 +15,8 @@ internal abstract class SuggestionModeCompletionProvider : CommonCompletionProvi
 
         public override async Task ProvideCompletionsAsync(CompletionContext context)
         {
-            if (context.Options.GetOption(CompletionControllerOptions.AlwaysShowBuilder))
-            {
-                var text = await context.Document.GetTextAsync(context.CancellationToken).ConfigureAwait(false);
-                context.SuggestionModeItem = this.CreateEmptySuggestionModeItem();
-            }
```
Comment (Pilchie) [inline src/Features/Core/Portable/Completion/SuggestionMode/SuggestionModeCompletionProvider.cs]: Why is this code not necessary anymore?

Comment (rchande) [inline src/Features/Core/Portable/Completion/SuggestionMode/SuggestionModeCompletionProvider.cs]: @Pilchie It was already dead--we weren't actually setting that option any where. The controller/model were already handling the "empty builder" case.

**Comment by rchande** [comment]
@Pilchie Do you think this would meet the bar for 2.0 or should I retarget to post-dev15?

**Comment by Pilchie** [comment]
`post-dev15` please.

**Pull request merged**